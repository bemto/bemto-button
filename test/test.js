import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import BemtoButton from '..';

const ExtendedButton = BemtoButton.extend`
  background: lime;

  ${BemtoButton.focusCSS(`
    box-shadow: 0 0 5px 5px pink;
  `)}
`;

const testSnapshot = function(tag, props, children) {
  const tree = renderer.create(
    React.createElement(
      tag,
      props,
      children
    )
  ).toJSON();

  expect(tree).toMatchSnapshot();
}

test('Just a button', () => {
  testSnapshot(
    BemtoButton,
    {},
    'Some text'
  );
});

test('Disabled button', () => {
  testSnapshot(
    BemtoButton,
    { disabled: true },
    'Some text'
  );
});

test('Disabled link', () => {
  testSnapshot(
    BemtoButton,
    { disabled: true, href: '#x' },
    'Some text'
  );
});

test('A button with modifier and both elements', () => {
  testSnapshot(
    BemtoButton,
    {
      href: '#x',
      _mod: true,
      __Before: '←',
      __After: '→',
    },
    'Some text'
  );
});

test('An extended button', () => {
  testSnapshot(
    ExtendedButton,
    {
      type: null
    },
    'Some text'
  );
});
