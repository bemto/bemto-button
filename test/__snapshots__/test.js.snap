// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A button with modifier and both elements 1`] = `
.c0 {
  -moz-appearance: none;
  position: relative;
  z-index: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
  vertical-align: baseline;
  overflow: visible;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0;
  border: none;
  margin: 0;
  border-radius: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background: transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0_disabled {
  pointer-events: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: none;
}

.c0__Content {
  position: relative;
  display: block;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: border-box;
  width: 100%;
  border-radius: inherit;
}

.c0__Focus {
  visibility: hidden;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: inherit;
}

.c0__BLHelper {
  float: left;
  width: 0;
  min-width: 0;
}

.c0__BLHelper::before {
  content: "\\a0";
}

.c0__Text {
  display: block;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  min-width: 0;
}

*:root .c0__Text,
.c0__Text:-ms-input-placeholder {
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  width: 100%;
}

.c0__Before {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: left;
}

.c0__After {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: right;
}

<a
  className="c0 sc-bdVaJa_mod c0_mod"
  href="#x"
  tabIndex="0"
>
  <span
    className="__Content c0__Content"
    tabIndex="-1"
  >
    <span
      className="__BLHelper c0__BLHelper"
    />
    <span
      className="__Before c0__Before"
    >
      ←
    </span>
    <span
      className="__After c0__After"
    >
      →
    </span>
    <span
      className="__Text c0__Text"
    >
      Some text
    </span>
  </span>
  <span
    className="__Focus c0__Focus"
  />
</a>
`;

exports[`An extended button 1`] = `
.c0 {
  -moz-appearance: none;
  position: relative;
  z-index: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
  vertical-align: baseline;
  overflow: visible;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0;
  border: none;
  margin: 0;
  border-radius: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background: transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: lime;
}

.c0_disabled {
  pointer-events: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: none;
}

.c0__Content {
  position: relative;
  display: block;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: border-box;
  width: 100%;
  border-radius: inherit;
}

.c0__Focus {
  visibility: hidden;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: inherit;
}

.c0__BLHelper {
  float: left;
  width: 0;
  min-width: 0;
}

.c0__BLHelper::before {
  content: "\\a0";
}

.c0__Text {
  display: block;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  min-width: 0;
}

*:root .c0__Text,
.c0__Text:-ms-input-placeholder {
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  width: 100%;
}

.c0__Before {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: left;
}

.c0__After {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: right;
}

.c0:focus,
.c0__Content:focus {
  outline: none;
}

*:focus > .c0__Focus {
  visibility: inherit;
  box-shadow: 0 0 5px 5px pink;
}

*:not(:-moz-focusring):focus > .c0__Focus {
  visibility: hidden;
}

<span
  className="c0"
  role="button"
  tabIndex="0"
  type={null}
>
  <span
    className="__Content c0__Content"
    tabIndex="-1"
  >
    <span
      className="__Text c0__Text"
    >
      Some text
    </span>
  </span>
  <span
    className="__Focus c0__Focus"
  />
</span>
`;

exports[`Disabled button 1`] = `
.c0 {
  -moz-appearance: none;
  position: relative;
  z-index: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
  vertical-align: baseline;
  overflow: visible;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0;
  border: none;
  margin: 0;
  border-radius: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background: transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0_disabled {
  pointer-events: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: none;
}

.c0__Content {
  position: relative;
  display: block;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: border-box;
  width: 100%;
  border-radius: inherit;
}

.c0__Focus {
  visibility: hidden;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: inherit;
}

.c0__BLHelper {
  float: left;
  width: 0;
  min-width: 0;
}

.c0__BLHelper::before {
  content: "\\a0";
}

.c0__Text {
  display: block;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  min-width: 0;
}

*:root .c0__Text,
.c0__Text:-ms-input-placeholder {
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  width: 100%;
}

.c0__Before {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: left;
}

.c0__After {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: right;
}

<button
  className="c0 sc-bdVaJa_disabled c0_disabled"
  disabled={true}
  tabIndex="-1"
  type="button"
>
  <span
    className="__Content c0__Content"
    tabIndex="-1"
  >
    <span
      className="__Text c0__Text"
    >
      Some text
    </span>
  </span>
  <span
    className="__Focus c0__Focus"
  />
</button>
`;

exports[`Disabled link 1`] = `
.c0 {
  -moz-appearance: none;
  position: relative;
  z-index: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
  vertical-align: baseline;
  overflow: visible;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0;
  border: none;
  margin: 0;
  border-radius: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background: transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0_disabled {
  pointer-events: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: none;
}

.c0__Content {
  position: relative;
  display: block;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: border-box;
  width: 100%;
  border-radius: inherit;
}

.c0__Focus {
  visibility: hidden;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: inherit;
}

.c0__BLHelper {
  float: left;
  width: 0;
  min-width: 0;
}

.c0__BLHelper::before {
  content: "\\a0";
}

.c0__Text {
  display: block;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  min-width: 0;
}

*:root .c0__Text,
.c0__Text:-ms-input-placeholder {
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  width: 100%;
}

.c0__Before {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: left;
}

.c0__After {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: right;
}

<a
  className="c0 sc-bdVaJa_disabled c0_disabled"
  disabled={true}
  href="#x"
  tabIndex="-1"
>
  <span
    className="__Content c0__Content"
    tabIndex="-1"
  >
    <span
      className="__Text c0__Text"
    >
      Some text
    </span>
  </span>
  <span
    className="__Focus c0__Focus"
  />
</a>
`;

exports[`Just a button 1`] = `
.c0 {
  -moz-appearance: none;
  position: relative;
  z-index: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-block;
  vertical-align: baseline;
  overflow: visible;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0;
  border: none;
  margin: 0;
  border-radius: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background: transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0_disabled {
  pointer-events: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: none;
}

.c0__Content {
  position: relative;
  display: block;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: border-box;
  width: 100%;
  border-radius: inherit;
}

.c0__Focus {
  visibility: hidden;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: inherit;
}

.c0__BLHelper {
  float: left;
  width: 0;
  min-width: 0;
}

.c0__BLHelper::before {
  content: "\\a0";
}

.c0__Text {
  display: block;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  min-width: 0;
}

*:root .c0__Text,
.c0__Text:-ms-input-placeholder {
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  width: 100%;
}

.c0__Before {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: left;
}

.c0__After {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  float: right;
}

<button
  className="c0"
  role="button"
  tabIndex="0"
  type="button"
>
  <span
    className="__Content c0__Content"
    tabIndex="-1"
  >
    <span
      className="__Text c0__Text"
    >
      Some text
    </span>
  </span>
  <span
    className="__Focus c0__Focus"
  />
</button>
`;
